src/Options/ParseOptions.hpp

```diff
@@ -0,0 +1,606 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines classes and functions that handle parsing of input parameters.
+
+#pragma once
+
+#include <cstring>
+#include <exception>
+#include <iterator>
+#include <map>
+#include <ostream>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+#include <yaml-cpp/yaml.h>
+
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Options/Options.hpp"
+#include "Options/OptionsDetails.hpp"
+#include "Utilities/Overloader.hpp"
+#include "Utilities/PrettyType.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/StdHelpers.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+/// \ingroup OptionParsing
+/// The type that options are passed around as.  Contains YAML node
+/// data and an OptionContext.
+class Option_t {
```

Comment:
The `_t` should be only used for denoting eager metafunctions, not classes

Reply:
Option_t -> Option